lib/api/apiUtils/object/locationKeysSanityCheck.js

```diff
@@ -0,0 +1,20 @@
+/**
+* Check keys that exist in the current list which will be used in composing
+* object. This method checks against accidentally removing data keys due to
+* instability from the metadata layer. The check returns true if there was no
+* match and false if at least one key from the previous list exists in the
+* current list
+* @param {array|string} prev - list of keys from the object being overwritten
+* @param {array} curr - list of keys to be used in composing current object
+* @returns {array} list of keys that can be deleted
+*/
+function locationKeysSanityCheck(prev, curr) {
+    // backwards compatibility check if object is of model version 2
+    if (typeof prev === 'string') {
+        return curr.every(v => v.key !== prev);
+    }
+    const keysList = prev.map(v => v.key);
+    return curr.every(v => !keysList.includes(v.key));
```

Comment:
Looks like this line has an O(n^2) complexity in the number of parts, I fear it could cause long lock-ups of the process with 1-10K MPUs being completed on top of an existing 1-10K MPU, as the node process could loop up to 100M times without giving hand to the event loop.

I suggest to store all prev keys in a js object first instead of using `Array.map`, to benefit from an efficient lookup.

Reply:
I debated it but I didn't think of the case where a 10K mpu could replace a 10K mpu.